LocalSettings.php

```diff
@@ -201,199 +213,344 @@
 $wgDefaultUserOptions['visualeditor-enable'] = 1;
 $wgDefaultUserOptions['visualeditor-editor'] = "visualeditor";
 $wgVirtualRestConfig['modules']['parsoid'] = [
-    'url' =>  'https://testwiki.wiki/rest.php',
-    'domain' => 'testwiki.wiki',
+	'url' => 'https://testwiki.wiki/rest.php',
+	'domain' => 'testwiki.wiki',
 ];
 $wgStaffPowersShoutWikiMessages = false;
 $wgTitleBlacklistSources = array(
   array(
-    'type' => 'localpage',
-    'src'  => 'MediaWiki:Titleblacklist'
+	'type' => 'localpage',
+	'src'  => 'MediaWiki:Titleblacklist'
   )
 );
-//MatomoAnalytics
+
+// MatomoAnalytics
 $wgMatomoAnalyticsServerURL = 'https://piwik.testwiki.wiki/';
 $wgMatomoAnalyticsSiteID = 3;
+
 // WikiEditor
 $wgDefaultUserOptions['usebetatoolbar'] = 1;
 $wgDefaultUserOptions['usebetatoolbar-cgd'] = 1;
 $wgDefaultUserOptions['wikieditor-preview'] = 0;
 $wgDefaultUserOptions['wikieditor-publish'] = 0;
+
 // User rights
-$wgGroupPermissions['*']['abusefilter-view'] = true;
+/// All
 $wgGroupPermissions['*']['abusefilter-log'] = true;
+$wgGroupPermissions['*']['abusefilter-view'] = true;
 $wgGroupPermissions['*']['oathauth-enable'] = false;
 $wgGroupPermissions['*']['patrolmarks'] = true;
+
+/// User
+$wgGroupPermissions['user']['createpage'] = false;
+$wgGroupPermissions['user']['createtalk'] = false;
+$wgGroupPermissions['user']['edit'] = false;
+$wgGroupPermissions['user']['movefile'] = false;
 $wgGroupPermissions['user']['move'] = false;
-$wgGroupPermissions['user']['move-subpages'] = false;
-$wgGroupPermissions['user']['move-rootuserpages'] = false; // can move root userpages
 $wgGroupPermissions['user']['move-categorypages'] = false;
-$wgGroupPermissions['user']['movefile'] = false;
-$wgGroupPermissions['user']['user'] = true;
+$wgGroupPermissions['user']['move-rootuserpages'] = false; // can move root userpages
+$wgGroupPermissions['user']['move-subpages'] = false;
+$wgGroupPermissions['user']['read'] = false;
 $wgGroupPermissions['user']['translate'] = true;
-$wgGroupPermissions['user']['translate-messagereview'] = true;
 $wgGroupPermissions['user']['translate-groupreview'] = true;
 $wgGroupPermissions['user']['translate-import'] = true;
-$wgGroupPermissions['user']['createtalk'] = false;
-$wgGroupPermissions['user']['createpage'] = false;
-$wgGroupPermissions['user']['edit'] = false;
-$wgGroupPermissions['user']['read'] = false;
+$wgGroupPermissions['user']['translate-messagereview'] = true;
+$wgGroupPermissions['user']['user'] = true;
 $wgGroupPermissions['user']['writeapi'] = false;
-$wgGroupPermissions['bot']['writeapi'] = false;
```

Comment:
This seems to be removed completely?

Reply:
Like mine? It's a bot! It MUST be able to write to the API! Is the purpose of a bot